compat/maven-compat/src/main/java/org/apache/maven/artifact/deployer/DefaultArtifactDeployer.java

```diff
@@ -106,9 +104,6 @@ public void deploy(
                 org.eclipse.aether.artifact.Artifact pomArtifact = new SubArtifact(mainArtifact, "", "pom");
                 pomArtifact = pomArtifact.setFile(projectArtifactMetadata.getFile());
                 request.addArtifact(pomArtifact);
-            } else if (metadata instanceof SnapshotArtifactRepositoryMetadata
-                    || metadata instanceof ArtifactRepositoryMetadata) {
-                // eaten, handled by repo system
```

Comment:
How should I preserve this information? Empty business logic as documentation feels overkill.